### PR TITLE
Ct 3924 amend workflow to specifications

### PR DESCRIPTION
**Pull request by willmcb**
## Description
Allows end to end workflow and the ability for Managers to close cases.

## Self-review checklist

* [x] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [x] (3) Tests passing
* [x] (4) Branch ready to be merged (not work in progress)
* [x] (5) No superfluous changes in diff
* [x] (6) No TODO's without new ticket numbers
* [x] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`
* [ ] (8) Data migration script is created if any of letter templates is changed

### Related JIRA tickets
[CT-3924](https://dsdmoj.atlassian.net/browse/CT-3924)

### Manual testing instructions

1. Create a new SAR IR and assign it
2. Log in as a responder and accept and clear the case
3. Log in as an approver take the case on and clear the response
4. Log back in as a responder and mark the case as sent
5. Log in as a manager and close the case

### CODE REVIEW

File: config/state_machine/configs/00_moj/60_sar_internal_review/30_sar_internal_review_trigger_workflow.yml [hunk 1]
```diff
@@ -79,6 +79,24 @@
...
 
+              responded:
+                add_message_to_case:
+                  after_transition: Workflows::Hooks#notify_responder_message_received
+                respond:
+                close:
+                  transition_to: closed
+                destroy_case:
+                edit_case:
+                  if: Workflows::Predicates#not_overturned?
```
Comment (ymao2) [inline config/state_machine/configs/00_moj/60_sar_internal_review/30_sar_internal_review_trigger_workflow.yml:90]: very minor things, for this workflow,  IMO, all those checking whether the case is overturned ico doesn't seem relevant to SAR IR

File: app/controllers/concerns/sar_internal_review_cases_params.rb [hunk 2]
```diff
@@ -24,4 +24,38 @@ def create_sar_internal_review_params
...
   end
 
+  def process_sar_closure_params
+    params.require(:sar_internal_review).permit(
+      :date_responded_dd,
+      :date_responded_mm,
+      :date_responded_yyyy,
+      :late_team_id,
+    ).merge(refusal_reason_abbreviation: missing_info_to_tmm)
+  end
```
Comment (ymao2) [inline app/controllers/concerns/sar_internal_review_cases_params.rb:34]: Not sure about this one,  do you actually use it?

**Review by ymao2 — COMMENTED** [review]
Very nice work! 👍 👍   only just 2 minor comments for your to double check

File: config/state_machine/configs/00_moj/60_sar_internal_review/30_sar_internal_review_trigger_workflow.yml (hunk 1 above)
Comment (willmcb) [reply config/state_machine/configs/00_moj/60_sar_internal_review/30_sar_internal_review_trigger_workflow.yml:90]: Thanks I will remove this in my next PR - I wasn't sure what was meant by overturned so I will remove it - as the next ticket is to set up editing etc.

File: app/controllers/concerns/sar_internal_review_cases_params.rb (hunk 2 above)
Comment (willmcb) [reply app/controllers/concerns/sar_internal_review_cases_params.rb:34]: Thanks I will take a look at remove if not :)

**Review by ymao2 — APPROVED** [review]
LGTM :)
